### PR TITLE
Adding test for correctness alongside benchmark + fixing reflection-based parser.

### PR DESCRIPTION
**Pull request by FranciscoThiesen**

### CODE REVIEW

File: benchmarks/src/benchmark_serialization_twitter.cpp [hunk 1]
```diff
@@ -27,12 +25,30 @@ struct User {
     int64_t followers_count;
     int64_t friends_count;
     int64_t statuses_count;
+
+    bool operator==(const User &other) const {
```
Comment (lemire) [inline benchmarks/src/benchmark_serialization_twitter.cpp]: This is fine, but I think it can be simplified.

**Comment by lemire** [comment]
Let me merge and maybe tweak again.

File: benchmarks/src/benchmark_serialization_twitter.cpp (hunk 1 above)
Comment (FranciscoThiesen) [reply benchmarks/src/benchmark_serialization_twitter.cpp]: What do you have in mind here?

**Comment by FranciscoThiesen** [comment]
> Let me merge and maybe tweak again.

@lemire Feel free to merge and then tweak, or elaborate on the suggestion here so I can perform the necessary changes.

**Comment by lemire** [comment]
On my todo.

**Comment by FranciscoThiesen** [comment]
@lemire also added nlohmann benchmarking for twitter data in this branch (:

**Pull request merged**